onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalInfo.java

```diff
@@ -451,31 +252,28 @@ private String computeTimeLabel(Context context) {
         }
     }
 
-    /**
-     * Computes the arrival status label from the delay (i.e., schedule deviation), where positive
-     * means the bus is running late and negative means the bus is running ahead of schedule
-     *
-     * @param delay schedule deviation, in minutes, for this vehicle where positive
-     *              means the bus is running late and negative means the bus is running ahead of
-     *              schedule
-     * @return the arrival status label based on the deviation
-     */
-    public static String computeArrivalLabelFromDelay(Resources res, long delay) {
-        if (delay > 0) {
-            // Arriving delayed
-            return res.getQuantityString(
-                    R.plurals.stop_info_arrive_delayed, (int) delay,
-                    delay);
-        } else if (delay < 0) {
-            // Arriving early
-            delay = -delay;
-            return res
-                    .getQuantityString(
-                            R.plurals.stop_info_arrive_early,
-                            (int) delay, delay);
+    private String computeNotifyText(Context context) {
+        final String routeName = mInfo.getShortName();
+        if (mEta <= 0) {
+            if (mIsArrival) {
+                return context.getString(R.string.trip_stat_gone_arrived, routeName);
+            } else {
+                return context.getString(R.string.trip_stat_gone_departed, routeName);
+            }
+        } else if (mEta < ONE_MINUTE) {
```

Comment:
Yes, I just looked at `ArrivalsListAdapterStyleA.initView()` and `SimpleArrivalListFragment.loadArrivalList()` both of them have the following logic:

~~~
        long eta = stopInfo.getEta();
        if (eta == 0) {
            etaView.setText(R.string.stop_info_eta_now);
            minView.setVisibility(View.GONE);
        } else {
            etaView.setText(String.valueOf(eta));
            minView.setVisibility(View.VISIBLE);
        }
~~~

So testing `mEta` equivalence to `0` is the proper way to see if the bus is arriving or departing "now".